python/lsst/dbb/buffmngrs/endpoint/ingester/ingester.py

```diff
@@ -215,14 +222,14 @@ def run(self):
                         status = Status.IGNORED
                         logger.debug("%s: %s; matched pattern(s): %s",
                                      path, message, ', '.join(matches))
-                if size == 0:
+                if status is None and not os.path.isfile(abspath):
+                    message = "no such file in the storage area"
+                    status = Status.INVALID
+                if status is None and size == 0:
```

Comment:
The above if's can't be just changed to elif because the pattern test isn't included in the test, but this one can (unless you'd rather stick to the pattern).  It really did bother me that the others couldn't be elif because only one of these should ever be true.  But with the pattern matching I couldn't figure out an easy way to do it without moving pattern matching to a function and calling the function in the condition.

Reply:
I'll keep the current pattern for now if you don't mind. I find the code easier to follow when each check is independent. And the penalty for not using `elif` here should be rather negligible.